addons/sale_crm/crm_lead.py

```diff
@@ -11,6 +13,39 @@ def _get_sale_amount_total(self, cr, uid, ids, fields, args, context=None):
             res[key] = value
         return res
 
+    def action_view_quotation(self, cr, uid, ids, context=None):
+    # '''
+    # This function returns an action that display existing quotations of given opportunity ids. It can either be a in a list or in a form view, if there is only one quotation to show.
+    # '''
+        mod_obj = self.pool.get('ir.model.data')
+        act_obj = self.pool.get('ir.actions.act_window')
+
+        result = mod_obj.get_object_reference(cr, uid, 'sale', 'action_quotations')
+        id = result and result[1] or False
+        result = act_obj.read(cr, uid, [id], context=context)[0]
+        #compute the number of quotation to display
+        quote_ids = []
+        for op in self.browse(cr, uid, ids, context=context):
+            quote_ids += [quotation.id for quotation in op.quotation_ids]
+        #choose the view_mode accordingly
+        if len(quote_ids)>1:
+            result['domain'] = "[('id','in',["+','.join(map(str, quote_ids))+"])]"
+        else:
+            res = mod_obj.get_object_reference(cr, uid, 'sale', 'view_order_form')
+            result['views'] = [(res and res[1] or False, 'form')]
+            result['res_id'] = quote_ids and quote_ids[0] or False
+        return result
+
+    def _quotation_exists(self, cursor, user, ids, name, arg, context=None):
+        res = {}
+        for opportunity in self.browse(cursor, user, ids, context=context):
+            res[opportunity.id] = False
+            if opportunity.quotation_ids:
+                res[opportunity.id] = True
+        return res
+
     _columns = {
-        'sale_amount_total': fields.function(_get_sale_amount_total, string="Total Amount Of Quotations", type="float")
+        'sale_amount_total': fields.function(_get_sale_amount_total, string="Total Amount Of Quotations", type="float"),
+        'quotation_ids': fields.many2many('sale.order', 'quotation_sale_order_rel', 'lead_id', 'order_id', 'Quotations', readonly=True, copy=False, help="This is the list of quotations that have been generated for this opportunity. The same opportunity may have created quotations several times."),
```

Comment:
Why defining a many2many where you could use the inverse relationship of the opportunity_id field defined on the sale order model (see sale_crm/sale_crm.py) ?